cardano-rosetta-server/src/server/db/queries/search-transactions-queries.ts

```diff
@@ -0,0 +1,194 @@
+import { OperationType, CatalystLabels } from '../../utils/constants';
+import { SearchFilters } from '../../models';
+
+const withLimitAndOffset = (query: string) => `
+    ${query} 
+    ORDER BY block.id DESC 
+    LIMIT $1 
+    OFFSET $2
+`;
+
+const withFilters = (query: string, filters: SearchFilters): string => {
+  const { maxBlock, operator, status } = filters;
+  let whereClause = '';
+  const conditions = [];
+  if (maxBlock) {
+    conditions.push(`block.block_no <= ${maxBlock}`);
+  }
+  if (status !== undefined || status !== null) {
+    conditions.push(`tx.valid_contract = ${status}`);
+  }
+  if (conditions.length > 0) {
+    whereClause = `WHERE ${conditions.join(` ${operator} `)}`;
+  }
+  return query.concat(whereClause);
+};
+
+const transactionQuery = `
+SELECT 
+    tx.hash,
+    tx.fee,
+    tx.size,
+    tx.valid_contract AS "validContract",
+    tx.script_size AS "scriptSize",
+    block.hash AS "blockHash",
+    block.block_no as "blockNo"
+`;
+
+const totalCountQuery = 'SELECT COUNT(1)';
```

Comment:
I think it's being used like this
```suggestion
const totalCountQuery = 'SELECT COUNT(1) AS "totalCount"';
```